<?xml version="1.0"?>
<!--
Author: Pedro Martins (martinspedro@av.it.pt)
Description:
-->

<launch>
   <!-- PARAMETERS -->
   <param name="use_sim_time" value="true" />

   <!-- ARGUMENTS -->
   <!-- Datasets -->
      <arg name="datasets_directory"        value="/media/martinspedro/Elements/"/>
      <arg name="dataset_folder_path"       value="$(arg datasets_directory)mine/CAMBADA/2019-08-28\ (\Setup\ A\)/Camera\ Calibration/" />
      <arg name="default_bag"               default="original_raw.bag" />


   <node pkg="tf" type="static_transform_publisher" name="LiDAR_to_world" args="0 0 0.8 0 0 0 1  velo_link world 100" />

   <!--
    <node pkg="tf" type="static_transform_publisher" name="magic" args="0 0 0 0 0 0 1 camera_color_left camera_link  100"  />
    <node pkg="tf" type="static_transform_publisher" name="magic2" args="0 0 0 0 0 0 1  velodyne velo_link   100"  />
    Select only one -->

   <!-- Call rosbag node to publish LIDAR data using clocked time. Loop enabled
   <node pkg="rosbag" type="play" name="Dataset" required="true"
         args="$(arg dataset_folder_path)$(arg default_bag) clock -l">
      </node> -->




     <!-- Point Cloud Visualizer
          Selec between opening simple point cloud visualizer or RVIZ
       -->
     <node pkg="sensor_fusion" type="point_cloud_coloring" name="Point_Cloud_Coloring" output="screen" >
         <remap from="/kitti/camera_color_left/image_raw" to="/camera/image_color" />
         <remap from="/kitti/camera_color_left/camera_info" to= "/camera/camera_info"  />
    </node>




</launch>
